src/query/plan/operator.cpp

```diff
@@ -2269,10 +2269,24 @@ UniqueCursorPtr Filter::MakeCursor(utils::MemoryResource *mem) const {
 
 std::vector<Symbol> Filter::ModifiedSymbols(const SymbolTable &table) const { return input_->ModifiedSymbols(table); }
 
+static std::vector<UniqueCursorPtr> MakeCursorVector(const std::vector<std::shared_ptr<LogicalOperator>> &ops,
+                                                     utils::MemoryResource *mem) {
+  std::vector<UniqueCursorPtr> cursors;
+  cursors.reserve(ops.size());
+
+  if (!ops.empty()) {
+    for (const auto &op : ops) {
+      cursors.push_back(op->MakeCursor(mem));
```

Comment:
```suggestion
      cursors.emplace_back(op->MakeCursor(mem));
```

Reply:
Just to note: with a temporary `push_back` does the same thing as `emplace_back` (move construct), as it has an overload with rvalue reference.